sql/src/main/java/io/crate/operation/collect/sources/ShardCollectSource.java

```diff
@@ -445,7 +448,10 @@ private CrateCollector getShardsCollector(RoutedCollectPhase collectPhase,
 
         return new RowsCollector(
             rowReceiver,
-            Iterables.transform(rows, Buckets.arrayToRowFunction()));
+            CompletableFuture.completedFuture(rows),
+            objects -> Iterables.transform(
+                objects,
+                (Function<Object, Row>) input -> Buckets.arrayToRowFunction().apply((Object[]) input)));
```

Comment:
I think the `Object[]` cast could be avoided if the Function signature is changed.